### PR TITLE
telemetry: Log error reason for failure to get SQL Metric

### LINKED ISSUE #40777: Logging is missing detail for "Error exists when getting the SQL Metric."
## Bug Report

Example logging:

```
[2023/01/26 10:38:30.918 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
[2023/01/26 10:39:30.920 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
[2023/01/26 10:40:30.922 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
[2023/01/26 10:41:30.931 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
```

This is for TiDB v6.5.0

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #40777 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

With this PR:
```
[2023/01/26 10:57:37.536 +01:00] [INFO] [data_window.go:257] ["Error exists when getting the SQL Metric."] [error="[domain:9009]Prometheus address is not set in PD and etcd"]
```

Without this PR:
```
[2023/01/26 10:38:30.918 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
[2023/01/26 10:39:30.920 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
[2023/01/26 10:40:30.922 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
[2023/01/26 10:41:30.931 +01:00] [INFO] [data_window.go:256] ["Error exists when getting the SQL Metric."]
```

### CODE REVIEW

File: telemetry/data_window.go [hunk 1]
```diff
@@ -253,7 +254,8 @@ func RotateSubWindow() {
 
 	err := readSQLMetric(time.Now(), &thisSubWindow.SQLUsage)
 	if err != nil {
-		logutil.BgLogger().Info("Error exists when getting the SQL Metric.")
+		logutil.BgLogger().Info("Error exists when getting the SQL Metric.",
+			zap.String("error", err.Error()))
```
Comment (breezewish) [inline telemetry/data_window.go:258]: How about just using `zap.Error(err)`?

Comment (dveeden) [reply telemetry/data_window.go]: Changed this on a few other places as well in f4a975dc48fe3281199a122242a5ec4418194598. Looks more consistent now.

```
$ jq 'select(.error) | select(.error | contains("Prometheus"))' tidb.log
```
```json
{
  "level": "INFO",
  "time": "2023/01/27 10:00:35.554 +01:00",
  "caller": "data_slow_query.go:67",
  "message": "Failed to get Slow Query Stats",
  "error": "[domain:9009]Prometheus address is not set in PD and etcd"
}
{
  "level": "INFO",
  "time": "2023/01/27 10:01:36.223 +01:00",
  "caller": "data_window.go:257",
  "message": "Error exists when getting the SQL Metric.",
  "error": "[domain:9009]Prometheus address is not set in PD and etcd"
}
{
  "level": "INFO",
  "time": "2023/01/27 10:02:36.225 +01:00",
  "caller": "data_window.go:257",
  "message": "Error exists when getting the SQL Metric.",
  "error": "[domain:9009]Prometheus address is not set in PD and etcd"
}
```

Comment (dveeden) [reply telemetry/data_window.go]: PTAL @breezewish @hawkingrei

**Review by breezewish — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**